packages/components/src/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,148 @@
+   * The number of items per page.
+   */
+  pageSize: number;
+  /**
+   * Handler that is called when the pagination active page changes.
+   */
+  onChange?: (page: number) => void;
+  /**
+   * Labels for the pagination controls.
+   */
+  controlLabels?: [string, string];
+}
+
+const _Pagination = ({
+  defaultPage = 1,
+  page,
+  totalItems,
+  pageSize,
+  onChange = () => {},
+  controlLabels,
+}: PaginationProps) => {
+  const [currentPage, setCurrentPage] = useState(page ?? defaultPage);
+  const totalPages = Math.ceil(totalItems / pageSize);
+
+  const handlePageChange = (newPage: number) => {
+    setCurrentPage(newPage);
+    onChange(newPage);
+  };
+
+  const { registerRef, keyboardProps, setNavigationItems, setFocusedItem } =
+    useKeyboardNavigation({
+      page: currentPage,
+      totalPages,
+      onChange: handlePageChange,
+    });
+
+  const pageRange = usePageRange({ currentPage, totalPages });
+
+  useEffect(() => {
+    const items = [
+      { type: NavigationTypes.Prev, value: currentPage - 1 },
+      ...pageRange.map(value => ({
+        type:
+          typeof value === 'number'
+            ? NavigationTypes.Page
+            : NavigationTypes.Ellipsis,
+        value,
+      })),
+      { type: NavigationTypes.Next, value: currentPage + 1 },
+    ];
+    setNavigationItems(items);
+  }, [pageRange, currentPage, setNavigationItems]);
+
+  useEffect(() => {
+    setFocusedItem({ type: NavigationTypes.Page, value: currentPage });
+  }, [currentPage, setFocusedItem]);
+
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === totalPages || totalPages === 0;
+
+  return (
+    <nav
+      className="flex items-center justify-center space-x-2"
+      aria-label={`Page ${currentPage} of ${totalPages}`}
+      {...keyboardProps}
+    >
+      <PaginationButton
+        onPress={() => handlePageChange(Math.max(1, currentPage - 1))}
+        aria-label="Page previous"
+        isDisabled={isFirstPage}
+        registerRef={ref =>
+          registerRef(NavigationTypes.Prev, currentPage - 1, ref)
+        }
+        controlLabel={controlLabels?.[0]}
+        position="left"
+      >
+        <ChevronLeft className="h-5 w-5" />
+      </PaginationButton>
+
+      <div className="flex items-center space-x-2">
+        {totalPages > 0 ? (
+          pageRange.map((pageNumber, index) =>
+            pageNumber === 'ellipsis' ? (
+              <Ellipsis key={`ellipsis-${index}`} />
+            ) : (
+              <PageButton
+                key={pageNumber}
+                page={pageNumber}
+                isSelected={pageNumber === currentPage}
+                onPress={() => handlePageChange(pageNumber)}
+                registerRef={ref =>
+                  registerRef(NavigationTypes.Page, pageNumber, ref)
+                }
+              />
+            )
+          )
+        ) : (
+          <PageButton key={1} page={1} isDisabled />
+        )}
+      </div>
+
+      <PaginationButton
```

Comment:
Maybe I would call it NavigationButton or something like this, I was at first a little bit confused about `PageButton` and `PaginationButton`.